java-generator/core/src/main/java/io/fabric8/java/generator/CRGeneratorRunner.java

```diff
@@ -0,0 +1,137 @@
+
+public class CRGeneratorRunner {
+
+    public void run(File source, File basePath) {
+        try (final KubernetesClient client = new DefaultKubernetesClient()) {
+            // Parse CRD with fabric8
+            CustomResourceDefinition crd =
+                    client.apiextensions().v1().customResourceDefinitions().load(source).get();
+
+            List<WritableCRCompilationUnit> writables =
+                    generate(crd, getPackage(crd.getSpec().getGroup()));
+
+            for (WritableCRCompilationUnit w : writables) {
+                w.writeAllJavaClasses(basePath);
+            }
+        }
+    }
+
+    public List<WritableCRCompilationUnit> generate(
+            CustomResourceDefinition crd, Optional<String> basePackageName) {
+        CustomResourceDefinitionSpec crSpec = crd.getSpec();
+        String crName = crSpec.getNames().getKind();
+        String group = crSpec.getGroup();
+
+        List<WritableCRCompilationUnit> writableCUs =
+                new ArrayList<WritableCRCompilationUnit>(crSpec.getVersions().size());
+        for (CustomResourceDefinitionVersion crdv : crSpec.getVersions()) {
+            CompilationUnit cu = new CompilationUnit();
+
+            String version = crdv.getName();
+
+            String pkg = basePackageName.map((p) -> p + "." + version).orElse(version);
+
+            cu.setPackageDeclaration(pkg);
+
+            AbstractJSONSchema2Pojo specGenerator = null;
+
+            if (crdv.getSchema().getOpenAPIV3Schema().getProperties().get("spec") != null) {
+                specGenerator =
+                        AbstractJSONSchema2Pojo.fromJsonSchema(
+                                "spec",
+                                crdv.getSchema().getOpenAPIV3Schema().getProperties().get("spec"),
+                                crName,
+                                "");
+            }
+
+            AbstractJSONSchema2Pojo statusGenerator = null;
+            if (crdv.getSchema().getOpenAPIV3Schema().getProperties().get("status") != null) {
+                statusGenerator =
+                        AbstractJSONSchema2Pojo.fromJsonSchema(
+                                "status",
+                                crdv.getSchema().getOpenAPIV3Schema().getProperties().get("status"),
+                                crName,
+                                "");
+            }
+
+            AbstractJSONSchema2Pojo crGenerator =
+                    new JCRObject(
+                            crName, version, group, specGenerator != null, statusGenerator != null);
+
+            List<String> classNames = new ArrayList<String>();
+
+            classNames.addAll(
+                    validateAndAggregate(cu, crGenerator, specGenerator, statusGenerator));
+
+            writableCUs.add(new WritableCRCompilationUnit(cu, classNames));
+        }
+
+        return writableCUs;
+    }
+
+    private List<String> validateAndAggregate(CompilationUnit cu, AbstractJSONSchema2Pojo... grs) {
```

Comment:
Can you please use more descriptive parameter names?

Reply:
sure 👍